spec/features/users/sign_in_spec.rb

```diff
@@ -151,7 +151,7 @@ def warning_content
       allow(Rails.application.config).to receive(:session_check_delay).and_return(1)
       allow(Devise).to receive(:timeout_in).and_return(1.second)
 
-      user = sign_in_and_2fa_user
```

Comment:
lgtm. (although it would maybe be nice if sign_in_and_2fa_user and sign_in_user we named in such as way that the warden vs non-warden distinction was clear).